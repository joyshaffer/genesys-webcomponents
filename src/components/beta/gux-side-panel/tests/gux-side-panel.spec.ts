import { newSpecPage } from '@stencil/core/testing';
import { GuxSidePanel } from '../gux-side-panel';

const components = [GuxSidePanel];
const language = 'en';

describe('gux-side-panel', () => {
  it('should build', async () => {
    const html = `<gux-side-panel-beta></gux-side-panel-beta>`;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxSidePanel);
  });

  describe('#render', () => {
    [
      { position: 'left', isOpen: false },
      { position: 'left', isOpen: true },
      { position: 'right', isOpen: false },
      { position: 'right', isOpen: true }
    ].forEach(({ position, isOpen }, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const html = `<gux-side-panel-beta position="${position}" is-open="${isOpen}"></gux-side-panel-beta>`;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
