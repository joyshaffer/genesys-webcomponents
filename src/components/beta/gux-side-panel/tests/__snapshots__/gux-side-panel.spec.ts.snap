// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-side-panel #render should render component as expected (1) 1`] = `
<gux-side-panel-beta is-open="false" position="left">
  <!---->
  <aside class="gux-closed gux-left">
    <div class="gux-panel-icons"></div>
    <div class="gux-closed gux-panel-content"></div>
  </aside>
</gux-side-panel-beta>
`;

exports[`gux-side-panel #render should render component as expected (2) 1`] = `
<gux-side-panel-beta is-open="true" position="left">
  <!---->
  <aside class="gux-left gux-open">
    <div class="gux-panel-icons"></div>
    <div class="gux-open gux-panel-content"></div>
  </aside>
</gux-side-panel-beta>
`;

exports[`gux-side-panel #render should render component as expected (3) 1`] = `
<gux-side-panel-beta is-open="false" position="right">
  <!---->
  <aside class="gux-closed gux-right">
    <div class="gux-panel-icons"></div>
    <div class="gux-closed gux-panel-content"></div>
  </aside>
</gux-side-panel-beta>
`;

exports[`gux-side-panel #render should render component as expected (4) 1`] = `
<gux-side-panel-beta is-open="true" position="right">
  <!---->
  <aside class="gux-open gux-right">
    <div class="gux-panel-icons"></div>
    <div class="gux-open gux-panel-content"></div>
  </aside>
</gux-side-panel-beta>
`;
